runtime/restart/monitor/monitor.go

```diff
@@ -218,19 +218,29 @@ func (m *monitor) monitor(ctx context.Context) ([]change, error) {
 		if err != nil {
 			return nil, err
 		}
+		var status containerd.Status
 		desiredStatus := containerd.ProcessStatus(labels[restart.StatusLabel])
 		task, err := c.Task(ctx, nil)
-		if err != nil && desiredStatus == containerd.Stopped {
+		if err != nil {
+			logrus.WithError(err).Error("monitor")
+			if desiredStatus == containerd.Stopped {
+				goto desiredSwitch
```

Comment:
Let's see if we can avoid `goto` and rewrite the logic a bit more clearly.

```go
var status containerd.Status
if task, err := c.Task(ctx, nil); err == nil {
	status, err = task.Status(ctx)
	if err == nil {
		if desiredStatus == status.Stopped {
			continue
		}
	} else {
		logrus.WithError(err).Error("monitor")
		if desiredStatus == containerd.Stopped {
			continue
		}
	}
} else {
	logrus.WithError(err).Error("monitor")
	if desiredStatus == containerd.Stopped {
		continue
	}
}
```

This allows error cases where `desiredStatus` is equal to `containerd.Running` to fall through to the switch statement while avoiding the scope clarity issues with goto.

Reply:
How about this:
```
		if task, err = c.Task(ctx, nil); err == nil {
			if status, err = task.Status(ctx); err == nil {
				if desiredStatus == status.Status {
					continue
				}
			}
		}

		// Task or Status return error, only desired to stop
		if err != nil {
			logrus.WithError(err).Error("monitor")
			if desiredStatus != containerd.Stopped {
				continue
			}
		}
```